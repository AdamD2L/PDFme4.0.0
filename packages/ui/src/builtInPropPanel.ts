import type { PropPanel } from './types';
import { barcodeSchemaTypes } from '@pdfme/common';
import { text, image, barcodes } from "@pdfme/schemas"

const propPanel: PropPanel = {
    text: text.propPanel(),
    image: image.propPanel(),
    ...barcodeSchemaTypes.reduce((acc, type) =>
        Object.assign(acc, {
            [type]: barcodes.propPanel(type)
        }), {}),
}
export default propPanel