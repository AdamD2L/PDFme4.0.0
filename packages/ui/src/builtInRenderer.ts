import { barcodeSchemaTypes } from '@pdfme/common';
import type { Renderer } from './types';
import { renderText } from './renders/text';
import { renderImage } from './renders/image';
import { renderBarcode } from './renders/barcodes';

const barcodeDefaults = {
    qrcode: { size: { width: 30, height: 30, }, value: 'https://pdfme.com/' },
    japanpost: { size: { width: 80, height: 7.2, }, value: '6540123789-A-K-Z' },
    ean13: { size: { width: 40, height: 16, }, value: '2112345678900' },
    ean8: { size: { width: 40, height: 20, }, value: '02345673' },
    code39: { size: { width: 40, height: 20, }, value: 'THIS IS CODE 39' },
    code128: { size: { width: 40, height: 20, }, value: 'This is Code 128!' },
    nw7: { size: { width: 40, height: 20, }, value: 'A0123456789B' },
    itf14: { size: { width: 40, height: 12, }, value: '04601234567893' },
    upca: { size: { width: 40, height: 16, }, value: '416000336108' },
    upce: { size: { width: 40, height: 20, }, value: '00123457' },
    gs1datamatrix: { size: { width: 30, height: 30, }, value: '(01)03453120000011(17)191125(10)ABCD1234' }
}

const renderer: Renderer = {
    text: {
        defaultValue: 'Text',
        defaultSize: { width: 40, height: 10 },
        render: renderText,
    },
    image: {
        defaultValue: 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAu4AAALuAQMAAADL0wGJAAAABGdBTUEAALGPC/xhBQAAAAFzUkdCAK7OHOkAAAAGUExURbzAw+rv8fKruy0AAAPoSURBVHja7dwxbtwwEEBRCkKwRQodYftcYk+ROkcJz5NTsEuZK/AIKlIQAUEnke0VqQ0pA5zxWvFnZcD2s0CNuENxPOZBc0QDDw8PDw8PDw8PDw8PDw8PDw8PDw8PDw8PDw8PDw8PDw8PDw8PDw8PDw8PDw8PDw8PDw8PDw8PDw8PDw8PDw8PDw8PDw8PDw8PDw8PDw8PDw8PDw8PDw8PDw8PDw8PDw8PD/+P8bkxvnTzzjTG0M2b5rh08rHNT518aPOnTt63+aGTd23edPJ2h//ax+/oO6Gzx6c78+cuPu7x01vmwx5/6uLnO/PjO+b/rGifvtWH3VnT9vmh/e3eqx/bc9d79af2YwEPDw8P/6r8r1GVt5VcUoZPtXxGhp9rGYEM72vbEBne1hJ5Gb6ayIvwqZppi/CxmgqL8KGaTIrwc3WDfATeV/ffIryr7r+PwNvq/vsIfH17D5/qbw/gb/mLKp/OqnycFPj14yqcJPm4vfp5VJ0cP6jyzqjy9rr2q/Dr4qzBp3V5E1wxp/V3rl8LXn32qF6fAY31Psv2NXi/5lQaH+Vuzak0eLsmPRppVPa3FPiU3QiFFDZmD4FCAh6yxV+Bn7O9isLmx2d/TGHr5rI7obDxtFkcKWyb89M2+U1/sX7Kv7Io8gaxFy5l2D/faRk++3x6PgsalV52+fwUW4j/+eGhDPuneyH/otHmp9jyfHHMLM6n4phZnI/FObA4H4qDWnF+Lo46xXlflG+I864o3xDnbVG+Ic6X9RXSfCrrK6T5WBZASPOhrFCQ5ufyjF+a92XdkjTvyrolad6WdUvS/KawSI7/sQn7JfDl+O+bsF8CX44fN2FvHnNnIT4Nm7BfAl+ON5uwXwJfjA/LCuk2BXvCvN0U7InxflmAtxV1gvx0U2N3luPtdZOlwj/FoR5vbuq85Pi48F6LD0scOk3+sq1cleP9ohlNfkpq/N9pGaMuH7T4ZVqGWYt/nBavyxstftblfaVIWJU/y72yODCfzJH5oMvPh+adLm+PzCdzZD5U/61ClT9Lnvyo8e7QvFHlE3ydj0zOu5ucickhcpgcJofJedUVk8j5b/nGgIeHh4d/I3x/RwLlhgfNH3DavSwGXf7YjT76uqCEQ/P37p9z7uKVmwspd17S7hul3PXK6fbsmnvWhO6GZlMnn3ri8gXN5GzHnX0B35ydj91814CHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHl+d/A9cKjmiL040TAAAAAElFTkSuQmCC',
        defaultSize: { width: 40, height: 40 },
        render: renderImage,
    },
    ...barcodeSchemaTypes.reduce((acc, barcodeType) => Object.assign(acc, {
        [barcodeType]: {
            defaultValue: barcodeDefaults[barcodeType].value,
            defaultSize: barcodeDefaults[barcodeType].size,
            render: renderBarcode
        }
    }), {}),
}
export default renderer