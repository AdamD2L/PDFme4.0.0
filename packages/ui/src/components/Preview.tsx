import React, { useCallback, useRef, useState, useEffect } from 'react';
import type { SchemaForUI, PreviewProps, Size } from '@pdfme/common';
import UnitPager from './UnitPager';
import Root from './Root';
import ErrorScreen from './ErrorScreen';
import CtlBar from './CtlBar';
import Paper from './Paper';
import Renderer from './Renderer';
import { useUIPreProcessor, useScrollPageCursor } from '../hooks';
import { templateSchemas2SchemasList, getPagesScrollTopByIndex } from '../helper';
import { theme } from 'antd';

const Preview = ({
  template,
  inputs,
  size,
  onChangeInput,
}: Omit<PreviewProps, 'domContainer'> & {
  onChangeInput?: (args: { index: number; value: string; key: string }) => void;
  size: Size;
}) => {
  const { token } = theme.useToken();

  const containerRef = useRef<HTMLDivElement>(null);
  const paperRefs = useRef<HTMLDivElement[]>([]);

  const [unitCursor, setUnitCursor] = useState(0);
  const [pageCursor, setPageCursor] = useState(0);
  const [zoomLevel, setZoomLevel] = useState(1);
  const [schemasList, setSchemasList] = useState<SchemaForUI[][]>([[]] as SchemaForUI[][]);

  const { backgrounds, pageSizes, scale, error } = useUIPreProcessor({ template, size, zoomLevel });

  const init = useCallback(async () => {
    const sl = await templateSchemas2SchemasList(template);
    setSchemasList(sl);
  }, [template]);

  useEffect(() => {
    if (unitCursor > inputs.length - 1) {
      setUnitCursor(inputs.length - 1);
    }
  }, [inputs]);

  useEffect(() => {
    void init();
  }, [init]);

  useScrollPageCursor({
    ref: containerRef,
    pageSizes,
    scale,
    pageCursor,
    onChangePageCursor: (p) => setPageCursor(p),
  });

  const handleChangeInput = ({ key, value }: { key: string; value: string }) =>
    onChangeInput && onChangeInput({ index: unitCursor, key, value });

  const isForm = Boolean(onChangeInput);

  const input = inputs[unitCursor];

  if (error) {
    return <ErrorScreen size={size} error={error} />;
  }

  return (
    <Root size={size} scale={scale}>
      <CtlBar
        size={size}
        pageCursor={pageCursor}
        pageNum={schemasList.length}
        setPageCursor={(p) => {
          if (!containerRef.current) return;
          containerRef.current.scrollTop = getPagesScrollTopByIndex(pageSizes, p, scale);
          setPageCursor(p);
        }}
        zoomLevel={zoomLevel}
        setZoomLevel={setZoomLevel}
      />
      <UnitPager
        size={size}
        unitCursor={unitCursor}
        unitNum={inputs.length}
        setUnitCursor={setUnitCursor}
      />
      <div ref={containerRef} style={{ ...size, position: 'relative', overflow: 'auto' }}>
        <Paper
          paperRefs={paperRefs}
          scale={scale}
          size={size}
          schemasList={schemasList}
          pageSizes={pageSizes}
          backgrounds={backgrounds}
          renderSchema={({ schema, index }) => {
            const { key, readOnly } = schema;
            const content = readOnly ? schema.content || '' : (input && input[key]) || '';
            return (
              <Renderer
                key={schema.id}
                schema={schema}
                value={content}
                mode={isForm ? 'form' : 'viewer'}
                placeholder={schema.content}
                tabIndex={index + 100}
                onChange={(arg) => {
                  const args = Array.isArray(arg) ? arg : [arg];
                  args.forEach(({ key: _key, value }) => {
                    if (_key === 'content') {
                      handleChangeInput({ key, value: value as string });
                    } else {
                      const targetSchema = schemasList[pageCursor].find((s) => s.id === schema.id);
                      if (!targetSchema || !targetSchema[_key]) return;
                      targetSchema[_key] = value as string;
                      setSchemasList([...schemasList]);
                    }
                  });
                }}
                outline={
                  isForm && !schema.readOnly ? `1px dashed ${token.colorPrimary}` : 'transparent'
                }
                pageSize={pageSizes[pageCursor]}
                scale={scale}
              />
            );
          }}
        />
      </div>
    </Root>
  );
};

export default Preview;
