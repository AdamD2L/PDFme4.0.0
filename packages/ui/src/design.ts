import { ThemeConfig } from 'antd';

export const theme: ThemeConfig = {
  token: {
    fontSize: 14,
    margin: 8,
    marginLG: 12,
    marginXS: 4,
    padding: 8,
    paddingLG: 12,
    paddingXS: 4,
  },
  components: {
    Form: {
      fontSize: 12,
      itemMarginBottom: 4,
      verticalLabelPadding: '0 0 2px',
    },
  },
};

const background = 'rgb(74, 74, 74)';
const controllerBackground = '#777777bd';
const controllerTextColor = '#ffffff';
const controllerStyle = {
  textColor: controllerTextColor,
  background: controllerBackground,
  icon: {
    color: controllerTextColor,
    size: 19,
  },
  height: 38,
};
export const style = {
  ErrorScreen: {
    background,
    cardBackground: '#ffffffad',
  },
  Root: {
    background,
  },
  UnitPager: controllerStyle,
  CtlBar: { ...controllerStyle, barWidth: 360 },
};
