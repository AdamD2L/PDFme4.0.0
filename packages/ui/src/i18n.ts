import type { Lang, Dict } from '@pdfme/common';
import { DEFAULT_LANG } from './constants.js';

const dictEn = {
  cancel: 'Cancel',
  field: 'field',
  fieldName: 'Name',
  align: 'Align',
  width: 'Width',
  height: 'Height',
  rotate: 'Rotate',
  edit: 'Edit',
  plsInputName: 'Please input name',
  fieldMustUniq: 'Name of field is not unique',
  notUniq: '(Not unique name)',
  noKeyName: 'No name',
  fieldsList: 'Field List',
  addNewField: 'Add new field',
  editField: 'Edit Field',
  type: 'Type',
  errorOccurred: 'An error occurred',
  errorBulkUpdateFieldName:
    'Cannot commit the change because the number of items has been changed.',
  commitBulkUpdateFieldName: 'Commit Changes',
  bulkUpdateFieldName: 'Bulk update field names',
  barColor: 'Bar Color',
  textColor: 'Text Color',
  bgColor: 'Background Color',
  fontName: 'xxx',
  size: 'xxx',
  spacing: 'xxx',
  textAlign: 'xxx',
  verticalAlign: 'xxx',
  lineHeight: 'xxx',
  min: 'xxx',
  max: 'xxx',
  fit: 'xxx',
};

const dictJa: { [key in keyof Dict]: string } = {
  cancel: 'キャンセル',
  field: '入力項目',
  fieldName: '項目名',
  align: '整列',
  width: '幅',
  height: '高さ',
  rotate: '回転',
  edit: '編集する',
  plsInputName: '項目名を入力してください',
  fieldMustUniq: '他の入力項目名と被っています',
  notUniq: '(他の項目名と重複しています)',
  noKeyName: '項目名なし',
  fieldsList: '入力項目一覧',
  addNewField: '入力項目を追加',
  editField: '入力項目を編集',
  type: 'タイプ',
  errorOccurred: 'エラーが発生しました',
  errorBulkUpdateFieldName: '項目数が変更されているため変更をコミットできません。',
  commitBulkUpdateFieldName: '変更を反映',
  bulkUpdateFieldName: '項目名を一括変更',
  barColor: 'バーの色',
  textColor: 'テキストの色',
  bgColor: '背景色',
  fontName: 'Font Name',
  size: 'Size',
  spacing: 'Spacing',
  textAlign: 'Text Align',
  verticalAlign: 'Vertical Align',
  lineHeight: 'Line Height',
  min: 'Min',
  max: 'Max',
  fit: 'Fit',
};

const dictAr: { [key in keyof Dict]: string } = {
  cancel: 'إلغاء',
  field: 'الحقل',
  fieldName: 'اسم الحقل',
  align: 'محاذاة',
  width: 'العرض',
  height: 'الارتفاع',
  rotate: 'تدوير',
  edit: 'تعديل',
  plsInputName: 'الرجاء إدخال الاسم',
  fieldMustUniq: 'يجب أن يكون الحقل فريداً',
  notUniq: '(غير فريد)',
  noKeyName: 'لا يوجد اسم للحقل',
  fieldsList: 'قائمة الحقول',
  addNewField: 'إضافة حقل جديد',
  editField: 'تعديل الحقل',
  type: 'النوع',
  errorOccurred: 'حدث خطأ',
  errorBulkUpdateFieldName: 'لا يمكن تنفيذ التغيير لأنه تم تغيير عدد العناصر.',
  commitBulkUpdateFieldName: 'تنفيذ التغييرات',
  bulkUpdateFieldName: 'تغيير الأسماء',
  barColor: 'لون الشريط',
  textColor: 'لون الخط',
  bgColor: 'لون الخلفية',
  // TODO 3 多言語化する
  fontName: 'xxx',
  size: 'xxx',
  spacing: 'xxx',
  textAlign: 'xxx',
  verticalAlign: 'xxx',
  lineHeight: 'xxx',
  min: 'xxx',
  max: 'xxx',
  fit: 'xxx',
};

const dictTh: { [key in keyof Dict]: string } = {
  cancel: 'ยกเลิก',
  field: 'ฟิลด์',
  fieldName: 'ชื่อฟิลด์',
  align: 'จัดเรียง',
  width: 'ความกว้าง',
  height: 'ความสูง',
  rotate: 'หมุน',
  edit: 'แก้ไข',
  plsInputName: 'กรุณาใส่ชื่อ',
  fieldMustUniq: 'ชื่อฟิลด์ต้องไม่ซ้ำกัน',
  notUniq: '(ชื่อฟิลด์ซ้ำกัน)',
  noKeyName: 'ไม่มีชื่อ',
  fieldsList: 'รายการฟิลด์ทั้งหมด',
  addNewField: 'เพิ่มฟิลด์ใหม่',
  editField: 'แก้ไขฟิลด์',
  type: 'ประเภท',
  errorOccurred: 'เกิดข้อผิดพลาด',
  errorBulkUpdateFieldName: 'ไม่สามารถยืนยันการแก้ไขได้เนื่องจากจำนวนรายการมีการเปลี่ยนแปลง',
  commitBulkUpdateFieldName: 'ยืนยันการแก้ไข',
  bulkUpdateFieldName: 'แก้ไขชื่อฟิลด์เป็นชุด',
  barColor: 'สีบาร์',
  textColor: 'สีข้อความ',
  bgColor: 'สีพื้นหลัง',
  fontName: 'xxx',
  size: 'xxx',
  spacing: 'xxx',
  textAlign: 'xxx',
  verticalAlign: 'xxx',
  lineHeight: 'xxx',
  min: 'xxx',
  max: 'xxx',
  fit: 'xxx',
};

const dictIt: { [key in keyof Dict]: string } = {
  cancel: 'Annulla',
  field: 'Campo',
  fieldName: 'Nome',
  align: 'Allinea',
  width: 'Larghezza',
  height: 'Altezza',
  rotate: 'Ruota',
  edit: 'Modifica',
  plsInputName: 'Inserisci il nome per favore',
  fieldMustUniq: 'Il nome del campo non è univoco',
  notUniq: '(Nome non univoco)',
  noKeyName: 'Nessun nome',
  fieldsList: 'Lista campi',
  addNewField: 'Aggiungi un campo',
  editField: 'Modifica campo',
  type: 'Tipo',
  errorOccurred: 'Riscontrato errore',
  errorBulkUpdateFieldName:
    'Non è possibile salvare le modifiche perché il numero di elementi è cambiato.',
  commitBulkUpdateFieldName: 'Salva cambiamenti',
  bulkUpdateFieldName: 'Modifica nomi campi in blocco',
  barColor: 'Colore barra',
  textColor: 'Colore testo',
  bgColor: 'Colore sfondo',
  fontName: 'xxx',
  size: 'xxx',
  spacing: 'xxx',
  textAlign: 'xxx',
  verticalAlign: 'xxx',
  lineHeight: 'xxx',
  min: 'xxx',
  max: 'xxx',
  fit: 'xxx',
};

const dictPl: { [key in keyof Dict]: string } = {
  cancel: 'Anuluj',
  field: 'pole',
  fieldName: 'Klucz pola',
  align: 'Wyrównanie',
  width: 'Szerokość',
  height: 'Wysokość',
  rotate: 'Obrót',
  edit: 'Edytuj',
  plsInputName: 'Wymagane wprowadzenie klucza pola',
  fieldMustUniq: 'Klucz pola nie jest unikalny',
  notUniq: '(Klucz pola nie jest unikalny)',
  noKeyName: 'Brak nazwy klucza pola',
  fieldsList: 'Lista pól',
  addNewField: 'Dodaj nowe pole',
  editField: 'Edytuj pole',
  type: 'Typ pola',
  errorOccurred: 'Wystąpił błąd',
  errorBulkUpdateFieldName: 'Nie można wprowadzić zmian ponieważ liczba elementów uległa zmianie.',
  commitBulkUpdateFieldName: 'Zaakceptuj zmiany',
  bulkUpdateFieldName: 'Masowo aktualizuj klucze pól',
  barColor: 'Kolor paska',
  textColor: 'Kolor tekstu',
  bgColor: 'Kolor tła',
  fontName: 'xxx',
  size: 'xxx',
  spacing: 'xxx',
  textAlign: 'xxx',
  verticalAlign: 'xxx',
  lineHeight: 'xxx',
  min: 'xxx',
  max: 'xxx',
  fit: 'xxx',
};

const dictionaries: { [key in Lang]: Dict } = {
  en: dictEn,
  ja: dictJa,
  ar: dictAr,
  th: dictTh,
  it: dictIt,
  pl: dictPl,
};

export const getDict = (lang: Lang) => dictionaries[lang] || dictionaries[DEFAULT_LANG];
