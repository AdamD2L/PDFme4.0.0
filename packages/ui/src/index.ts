import Designer from './Designer';
import Form from './Form';
import Viewer from './Viewer';

import {
  BLANK_PDF,
  DEFAULT_FONT_VALUE,
  checkTemplate,
  checkUIProps,
  checkPreviewProps,
  checkDesignerProps,
  checkGenerateProps,
  validateBarcodeInput,
} from '@pdfme/common';

import type {
  Lang,
  Size,
  SchemaType,
  BarCodeType,
  Schema,
  SchemaForUI,
  Font,
  BasePdf,
  Template,
  CommonProps,
  GeneratorOptions,
  GenerateProps,
  UIOptions,
  UIProps,
  PreviewProps,
  DesignerProps,
} from '@pdfme/common';

export {
  Designer,
  Viewer,
  Form,
  BLANK_PDF,
  DEFAULT_FONT_VALUE,
  checkTemplate,
  checkUIProps,
  checkPreviewProps,
  checkDesignerProps,
  checkGenerateProps,
  validateBarcodeInput,
};

export type {
  Lang,
  Size,
  SchemaType,
  BarCodeType,
  Schema,
  SchemaForUI,
  Font,
  BasePdf,
  Template,
  CommonProps,
  GeneratorOptions,
  GenerateProps,
  UIOptions,
  UIProps,
  PreviewProps,
  DesignerProps,
};
