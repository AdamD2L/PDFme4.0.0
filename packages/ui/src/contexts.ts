import { createContext } from 'react';
import { getDict } from './i18n';
import { DEFAULT_LANG } from './constants.js';
import { type Dict, getDefaultFont, type Plugins } from '@pdfme/common';
import { builtInPlugins } from '@pdfme/schemas';

export const I18nContext = createContext((key: keyof Dict) => getDict(DEFAULT_LANG)[key]);

export const FontContext = createContext(getDefaultFont());

export const PluginsRegistry = createContext<Plugins>(builtInPlugins);

export const OptionsContext = createContext({});
