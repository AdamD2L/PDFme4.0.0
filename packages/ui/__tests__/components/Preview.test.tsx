/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render, act, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import Preview from '../../src/components/Preview';
import { I18nContext, FontContext, RendererRegistry } from '../../src/contexts';
import { curriedI18n } from '../../src/i18n';
import { SELECTABLE_CLASSNAME } from '../../src/constants';
import { getDefaultFont } from '@pdfme/common';
import { setupUIMock, getSampleTemplate } from '../assets/helper';
import { text, image } from "@pdfme/schemas"
import type { UIRenderer, UIRender } from '../../src/types';

const renderer: UIRenderer = {
  text: text.ui as UIRender,
  image: image.ui as UIRender
};

test('Preview(as Viewer) snapshot', async () => {
  setupUIMock();
  let container: HTMLElement = document.createElement('a');
  act(() => {
    const { container: c } = render(
      <I18nContext.Provider value={curriedI18n('en')}>
        <FontContext.Provider value={getDefaultFont()}>
          <RendererRegistry.Provider value={renderer}>
            <Preview
              template={getSampleTemplate()}
              inputs={[{ field1: 'field1', field2: 'field2' }]}
              size={{ width: 1200, height: 1200 }}
            />
          </RendererRegistry.Provider>
        </FontContext.Provider>
      </I18nContext.Provider>
    );
    container = c;
  });

  await waitFor(() => Boolean(container?.getElementsByClassName(SELECTABLE_CLASSNAME)));
  expect(container.firstChild).toMatchSnapshot();
});

test('Preview(as Form) snapshot', async () => {
  setupUIMock();
  let container: HTMLElement = document.createElement('a');
  act(() => {
    const { container: c } = render(
      <I18nContext.Provider value={curriedI18n('en')}>
        <FontContext.Provider value={getDefaultFont()}>
          <RendererRegistry.Provider value={renderer}>
            <Preview
              template={getSampleTemplate()}
              inputs={[{ field1: 'field1', field2: 'field2' }]}
              size={{ width: 1200, height: 1200 }}
              onChangeInput={console.log}
            />
          </RendererRegistry.Provider>
        </FontContext.Provider>
      </I18nContext.Provider>
    );
    container = c;
  });

  await waitFor(() => Boolean(container?.getElementsByClassName(SELECTABLE_CLASSNAME)));
  expect(container.firstChild).toMatchSnapshot();
});
