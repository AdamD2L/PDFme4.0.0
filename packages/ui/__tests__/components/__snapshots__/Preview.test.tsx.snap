// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview(as Form) snapshot 1`] = `
<div
  style="position: relative; background: rgb(74, 74, 74); width: 1200px; height: 1200px;"
>
  <div
    style="margin: 0px auto; width: 1200px; height: 1200px;"
  >
    <div
      style="position: absolute; bottom: 6%; width: 1200px;"
    >
      <div
        style="display: flex; align-items: center; justify-content: center; position: relative; z-index: 1; left: calc(50% - 90px); width: 180px; height: 38px; box-sizing: border-box; padding: 12px; border-radius: 6px; background-color: rgba(119, 119, 119, 0.741);"
      >
        <div
          style="display: flex;"
        >
          <button
            class="ant-btn css-dev-only-do-not-override-amq5gd ant-btn-text"
            type="button"
          >
            <svg
              aria-hidden="true"
              color="#ffffff"
              fill="none"
              height="19"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              width="19"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19.5 12h-15"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            
          </button>
          <span
            class="ant-typography css-dev-only-do-not-override-amq5gd"
            style="color: rgb(255, 255, 255); font-size: 14px; margin: 8px;"
          >
            <strong>
              100
              %
            </strong>
          </span>
          <button
            class="ant-btn css-dev-only-do-not-override-amq5gd ant-btn-text"
            type="button"
          >
            <svg
              aria-hidden="true"
              color="#ffffff"
              fill="none"
              height="19"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              width="19"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 4.5v15m7.5-7.5h-15"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            
          </button>
        </div>
      </div>
    </div>
    <div
      style="width: 1200px; height: 1200px; position: relative; overflow: auto;"
    >
      <div
        style="transform: scale(1); transform-origin: top left; height: 1px; width: 1px;"
      >
        <div
          id="@pdfme/ui-paper0"
          style="font-family: 'Roboto'; top: 20px; left: 203.14960629450002px; position: relative; background-image: url(data:image/png;base64,a...); background-size: 793.700787411px 1122.5196850527px; width: 793.700787411px; height: 1122.5196850527px;"
        >
          <div>
            <div
              class="selectable"
              id="1"
              style="position: absolute; cursor: pointer; height: 56.692913386499995px; width: 377.95275591px; top: 75.590551182px; left: 75.590551182px; transform: rotate(0deg); outline: 1px dashed #4af;"
              title="field1"
            >
              <div
                style="height: 100%; width: 100%;"
              >
                <div
                  style="padding: 0px; resize: none; display: flex; flex-direction: column; justify-content: flex-start; width: 100%; height: 100%;"
                >
                  <textarea
                    placeholder="bb"
                    rows="1"
                    style="padding: 3.436640625px 0px 0px 0px; resize: none; outline: none; background-color: transparent; width: 100%; height: 100%; font-family: inherit; color: rgb(0, 0, 0); font-size: 30pt; letter-spacing: 0pt; line-height: 1em; text-align: left; white-space: pre-wrap; word-break: break-word;"
                    tabindex="100"
                  />
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              class="selectable"
              id="2"
              style="position: absolute; cursor: pointer; height: 151.181102364px; width: 377.95275591px; top: 132.2834645685px; left: 75.590551182px; transform: rotate(0deg); outline: 1px dashed #4af;"
              title="field2"
            >
              <div
                style="height: 100%; width: 100%;"
              >
                <div
                  style="width: 100%; height: 100%; background-image: none; background-size: 377.95275591px 151.181102364px;"
                >
                  <img
                    src="field2"
                    style="height: 100%; width: 100%; border-radius: 0;"
                  />
                  <button
                    style="position: absolute; top: 0px; left: 0px; z-index: 1; display: flex; justify-content: center; align-items: center; color: rgb(51, 51, 51); background: rgb(242, 242, 242); border-radius: 2px; border: 1px solid #767676; cursor: pointer; height: 24px; width: 24px;"
                  >
                    x
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Preview(as Viewer) snapshot 1`] = `
<div
  style="position: relative; background: rgb(74, 74, 74); width: 1200px; height: 1200px;"
>
  <div
    style="margin: 0px auto; width: 1200px; height: 1200px;"
  >
    <div
      style="position: absolute; bottom: 6%; width: 1200px;"
    >
      <div
        style="display: flex; align-items: center; justify-content: center; position: relative; z-index: 1; left: calc(50% - 90px); width: 180px; height: 38px; box-sizing: border-box; padding: 12px; border-radius: 6px; background-color: rgba(119, 119, 119, 0.741);"
      >
        <div
          style="display: flex;"
        >
          <button
            class="ant-btn css-dev-only-do-not-override-amq5gd ant-btn-text"
            type="button"
          >
            <svg
              aria-hidden="true"
              color="#ffffff"
              fill="none"
              height="19"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              width="19"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19.5 12h-15"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            
          </button>
          <span
            class="ant-typography css-dev-only-do-not-override-amq5gd"
            style="color: rgb(255, 255, 255); font-size: 14px; margin: 8px;"
          >
            <strong>
              100
              %
            </strong>
          </span>
          <button
            class="ant-btn css-dev-only-do-not-override-amq5gd ant-btn-text"
            type="button"
          >
            <svg
              aria-hidden="true"
              color="#ffffff"
              fill="none"
              height="19"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              width="19"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 4.5v15m7.5-7.5h-15"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            
          </button>
        </div>
      </div>
    </div>
    <div
      style="width: 1200px; height: 1200px; position: relative; overflow: auto;"
    >
      <div
        style="transform: scale(1); transform-origin: top left; height: 1px; width: 1px;"
      >
        <div
          id="@pdfme/ui-paper0"
          style="font-family: 'Roboto'; top: 20px; left: 203.14960629450002px; position: relative; background-image: url(data:image/png;base64,a...); background-size: 793.700787411px 1122.5196850527px; width: 793.700787411px; height: 1122.5196850527px;"
        >
          <div>
            <div
              class="selectable"
              id="1"
              style="position: absolute; cursor: pointer; height: 56.692913386499995px; width: 377.95275591px; top: 75.590551182px; left: 75.590551182px; transform: rotate(0deg); outline: transparent;"
              title="field1"
            >
              <div
                style="height: 100%; width: 100%;"
              >
                <div
                  style="padding: 0px; resize: none; background-color: transparent; display: flex; flex-direction: column; justify-content: flex-start; width: 100%; height: 100%;"
                >
                  <div
                    style="font-family: inherit; color: rgb(0, 0, 0); font-size: 30pt; letter-spacing: 0pt; line-height: 1em; text-align: left; white-space: pre-wrap; word-break: break-word; margin-bottom: 0px; padding-top: 3.436640625px;"
                  >
                    <span
                      style="letter-spacing:inherit;"
                    >
                      f
                    </span>
                    <span
                      style="letter-spacing:inherit;"
                    >
                      i
                    </span>
                    <span
                      style="letter-spacing:inherit;"
                    >
                      e
                    </span>
                    <span
                      style="letter-spacing:inherit;"
                    >
                      l
                    </span>
                    <span
                      style="letter-spacing:inherit;"
                    >
                      d
                    </span>
                    <span
                      style="letter-spacing:0;"
                    >
                      1
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              class="selectable"
              id="2"
              style="position: absolute; cursor: pointer; height: 151.181102364px; width: 377.95275591px; top: 132.2834645685px; left: 75.590551182px; transform: rotate(0deg); outline: transparent;"
              title="field2"
            >
              <div
                style="height: 100%; width: 100%;"
              >
                <div
                  style="width: 100%; height: 100%; background-image: none; background-size: 377.95275591px 151.181102364px;"
                >
                  <img
                    src="field2"
                    style="height: 100%; width: 100%; border-radius: 0;"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
